File: QuevedoVirtualClassrooms-Server/src/main/java/com/quevedo/virtualclassroomsserver/common/models/common/ResourceType.java
package com.quevedo.virtualclassroomsserver.common.models.common;

public enum ResourceType {
    VIDEO(1, "VIDEO"),
    IMAGE(2,"IMAGE"),
    URL(3, "URL");

    private int value;
    private String stringValue;

    ResourceType(int value, String stringValue){
        this.value = value;
        this.stringValue = stringValue;
    }

    public static ResourceType getTypeByString(String inputString){
        ResourceType toReturn;
        switch (inputString){
            case "VIDEO":
                toReturn = ResourceType.VIDEO;
                break;
            case "IMAGE":
                toReturn = ResourceType.IMAGE;
                break;
            case "URL":
                toReturn = ResourceType.URL;
                break;
            default:
                toReturn = null;
        }
        return toReturn;
    }

    public static ResourceType getTypeByInt(int inputInt){
        switch (inputInt){
            case 1:
                return ResourceType.VIDEO;
            case 2:
                return ResourceType.IMAGE;
            case 3:
                return ResourceType.URL;
            default:
                return null;
        }
    }
    public int getValue(){
        return value;
    }

    public String getStringValue(){ return stringValue; }
}

File: QuevedoVirtualClassrooms-Android/app/src/main/java/org/quevedo/quevedovirtualclassrooms/data/models/common/ResourceType.kt
package org.quevedo.quevedovirtualclassrooms.data.models.common


enum class ResourceType(val value: Int, val stringValue: String) {
    VIDEO(1, "VIDEO"), IMAGE(2, "IMAGE"), URL(3, "URL");

    companion object {
        fun getTypeByString(inputString: String?): ResourceType? {
            val toReturn: ResourceType? = when (inputString) {
                "VIDEO" -> VIDEO
                "IMAGE" -> IMAGE
                "URL" -> URL
                else -> null
            }
            return toReturn
        }

        fun getTypeByInt(inputInt: Int): ResourceType? {
            return when (inputInt) {
                1 -> VIDEO
                2 -> IMAGE
                3 -> URL
                else -> null
            }
        }
    }
}